### PR TITLE
fix(2d): can not update world position for sprite renderer when call …

### PR DESCRIPTION
**Pull request by singlecoder**
…bounds in the same frame

### Please check if the PR fulfills these requirements

- [ ] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Review by cptbtptpbcptdtptp — CHANGES_REQUESTED** [review]
SrpiteRenderer 克隆的时候需要考虑 _spriteDirty.

File: packages/core/src/2d/sprite/SpriteRenderer.ts [hunk 1]
```diff
@@ -154,9 +160,9 @@ export class SpriteRenderer extends Renderer {
     const { transform } = this.entity;
 
     // Update sprite data.
-    const localDirty = sprite._updateMeshData();
+    sprite._updateMeshData();
 
-    if (this._isWorldMatrixDirty.flag || localDirty || this._isContainDirtyFlag(DirtyFlag.Sprite)) {
+    if (this._isWorldMatrixDirty.flag || this._spriteDirty.flag || this._isContainDirtyFlag(DirtyFlag.Sprite)) {
```
Comment (GuoLei1990) [inline packages/core/src/2d/sprite/SpriteRenderer.ts:165]: If we use `this._spriteDirty.flag`, `this._isContainDirtyFlag(DirtyFlag.Sprite)` is no need!

File: packages/core/src/2d/sprite/SpriteRenderer.ts [hunk 2]
```diff
@@ -154,9 +160,9 @@ export class SpriteRenderer extends Renderer {
     const { transform } = this.entity;
 
     // Update sprite data.
-    const localDirty = sprite._updateMeshData();
+    sprite._updateMeshData();
```
Comment (GuoLei1990) [inline packages/core/src/2d/sprite/SpriteRenderer.ts:163]: Use `_updateMesh()` and delete `_updateMeshData()`

File: packages/core/src/2d/sprite/SpriteRenderer.ts [hunk 3]
```diff
@@ -40,9 +40,11 @@ export class SpriteRenderer extends Renderer {
   @assignmentClone
   private _cacheFlipY: boolean = false;
   @ignoreClone
-  private _dirtyFlag: number = DirtyFlag.All;
+  private _dirtyFlag: number = DirtyFlag.Flip;
```
Comment (GuoLei1990) [inline packages/core/src/2d/sprite/SpriteRenderer.ts:43]: I think the init value is 0

**Review by GuoLei1990 — APPROVED** [review]
+1

**Review by cptbtptpbcptdtptp — APPROVED** [review]
+1

**Pull request merged**